pkg/webhook/subscription.go

```diff
@@ -24,7 +24,7 @@ func (s *SubscriptionAdmission) Handle(ctx context.Context, req admission.Reques
 
 	// review should be for a subscription
 	subscription := &opv1a1.Subscription{}
-	if err := s.Decoder.Decode(req, subscription); err != nil {
+	if err := (*s.Decoder).Decode(req, subscription); err != nil {
```

Comment:
Why dereference to get a copy? If `s.Decoder` is a pointer then `s.Decode.Decode` should work

Reply:
In [CR v0.18](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission@v0.18.0#Decoder), `Decoder` is an interface and not a struct as it was in [CR v0.17.2](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/webhook/admission@v0.17.2#Decoder) -- this basically renders `s.Decoder.Decode` undefined.

Reply:
This means something else is wrong. An interface does not replace an implementation it just captures a contract and described on the link you attached above, that contract includes a Decode function. This means that you should be able to call decode on the Decoder interface which will invoke the decode function on the actual instance. 

If `s.Decoder.Decode` returns a different behavior than `(*s.Decoder).Decode` it is a very strong indication that something else is wrong here, which we need to resolve before allowing this to merge, or at least until we fully understand why it behaves this way

Reply:
seems there is a breaking change as per https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.18.0

Reply:
A breaking change is ok as long as we are following the migration guide. I don't believe the migration guide wants us to use a pointer to an interface, but I might be wrong

Reply:
Rebase allowed reverting this change